src/DDTrace/Integrations/LaravelProvider.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+namespace DDTrace\Integrations;
+
+use DDTrace;
+use DDTrace\Encoders\Json;
+use DDTrace\Tags;
+use DDTrace\Tracer;
+use DDTrace\Transport\Http;
+use Illuminate\Foundation\Http\Events\RequestHandled;
+use Illuminate\Pipeline\Pipeline;
+use Illuminate\Routing\Events\RouteMatched;
+use Illuminate\Support\Facades\Route;
+use Illuminate\Support\ServiceProvider;
+use Illuminate\View\Engines\CompilerEngine;
+use OpenTracing\GlobalTracer;
+
+use function DDTrace\Time\fromMicrotime;
+
+/**
+ * DataDog Laravel tracing provider. Use by installing the dd-trace library:
+ *
+ * composer require datadog/dd-trace
+ *
+ * And then load the provider in config/app.php:
+ *
+ *     'providers' => array_merge(include(base_path('modules/system/providers.php')), [
+ *        // 'Illuminate\Html\HtmlServiceProvider', // Example
+ *
+ *        'DDTrace\Integrations\LaravelProvider',
+ *        'System\ServiceProvider',
+ *   ]),
+ */
+class LaravelProvider extends ServiceProvider
+{
+    public function register()
+    {
+        if (!extension_loaded('ddtrace')) {
+            trigger_error('ddtrace extension required to load Laravel integration.', E_USER_WARNING);
+            return;
+        }
+
+        if (php_sapi_name() == 'cli') {
+            return;
+        }
+
+        // Creates a tracer with default transport and default encoders
+        $tracer = new Tracer(new Http(new Json()));
+
+        // Sets a global tracer (singleton). Also store it in the Laravel
+        // container for easy Laravel-specific use.
+        GlobalTracer::set($tracer);
```

Comment:
My approach here was to keep all of the integration code consistent, especially since the Laravel integration uses several others - PDO, Memcached, Predis, etc. - that can also be used outside of Laravel or from Symfony. So we need GlobalTracer or another solution that's consistent between each integration. I added the tracer to the Laravel container to make it easy to write extensions or add instrumentation inside of controllers or whatnot in a Laravel-ish way.